docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,898 @@
+// Create a file object from a string path
+myFile = file('data/sampleA_rep1_normal_R1_001.fastq.gz')
+
+// Print file attributes
+println "File name: ${myFile.name}"
+println "Simple name: ${myFile.simpleName}"
+println "Extension: ${myFile.extension}"
+println "Parent directory: ${myFile.parent}"
+```
+
+Run the workflow:
+
+```bash
+nextflow run main.nf
+```
+
+You'll see various file attributes printed to the console:
+
+```console title="File Attributes Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [ecstatic_ampere] DSL2 - revision: f3fa3dcb48
+
+File name: sampleA_rep1_normal_R1_001.fastq.gz
+Simple name: sampleA_rep1_normal_R1_001
+Extension: gz
+Parent directory: /workspaces/training/side-quests/working_with_files/data
+```
+
+### Takeaway
+
+- The `file()` method creates a Path object from a string
+- You can access file metadata like name, extension, and path
+
+---
+
+## 2. Using Channels for File Handling
+
+While the `file()` method is useful for simple file operations, channels provide a more powerful way to handle multiple files in workflows. We could load our file into a channel using [`Channel.of()`](https://www.nextflow.io/docs/latest/reference/channel.html#of), but we have a much more convenient tool called [`Channel.fromPath()`](https://www.nextflow.io/docs/latest/reference/channel.html#frompath) which finds every file in a directory that matches a given pattern.
+
+### 2.1 Reading Files with Channel.fromPath
+
+Update your `main.nf` file:
+
+_Before_:
+
+```groovy title="main.nf" linenums="2"
+// Create a file object from a string path
+myFile = file('data/sampleA_rep1_normal_R1_001.fastq.gz')
+
+// Print file attributes
+println "File name: ${myFile.name}"
+println "Simple name: ${myFile.simpleName}"
+println "Extension: ${myFile.extension}"
+println "Parent directory: ${myFile.parent}"
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="2"
+// Reading files with Channel.fromPath
+ch_fastq = Channel.fromPath('data/sampleA_rep1_normal_R1_001.fastq.gz')
+ch_fastq.view { "Found file: $it" }
+
+// // Print file attributes
+// Comment these out for now, we'll come back to them!
+// println "File name: ${myFile.name}"
+// println "Simple name: ${myFile.simpleName}"
+// println "Extension: ${myFile.extension}"
+// println "Parent directory: ${myFile.parent}"
+```
+
+Run the workflow:
+
+```bash
+nextflow run main.nf
+```
+
+You'll see each file path being emitted as a separate element in the channel:
+
+```console title="Channel.fromPath Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [grave_meucci] DSL2 - revision: b09964a583
+
+Found file: /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz
+```
+
+Note how Nextflow has grabbed all the files in the `data` directory that match the pattern `*.fastq.gz` and turned them into `file` objects. Channel.fromPath() is a convenient way of creating a new channel populated by a list of files.
```

Comment:
based on the rest of the training, I am not sure which of the two options you wanted to go with here: I think we either need to update the code or the description